### PR TITLE
cache: fixes consistency with CloseOnContextDone and listeners

### PR DESCRIPTION
**Pull request by mathetake**
This fixes the compilation cache consistency with respect to the 
listeners and CloseOnContextDone configurations by including them
in the wasm.Module.ID.

### CODE REVIEW

File: internal/integration_test/README.md
Comment (mathetake) [inline internal/integration_test/README.md:1]: realized that this file is stale, so I just deleted while here.

File: internal/integration_test/filecache/filecache_test.go
Comment (mathetake) [inline internal/integration_test/filecache/filecache_test.go:1]: the tests in this file are the ones we lacked.

File: internal/wasm/module.go [hunk 1]
```diff
@@ -191,9 +192,24 @@ const (
 	MaximumTableIndex    = uint32(1 << 27)
 )
 
-// AssignModuleID calculates a sha256 checksum on `wasm` and set Module.ID to the result.
-func (m *Module) AssignModuleID(wasm []byte) {
-	m.ID = sha256.Sum256(wasm)
+// AssignModuleID calculates a sha256 checksum on `wasm` and other args, and set Module.ID to the result.
+// See the doc on Module.ID on what it's used for.
+func (m *Module) AssignModuleID(wasm []byte, withListener, withEnsureTermination bool) {
```
Comment (mathetake) [inline internal/wasm/module.go:197]: this is the essential change and others are just misc

**Review by codefromthecrypt — APPROVED** [review]
Thanks for the lightning turnaround!

**Pull request merged**